File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/data/api/services/ApiConfigurationService.java
package com.crpr.androidcinema.data.api.services;

import com.crpr.androidcinema.data.api.models.configuration.ApiConfiguration;
import com.crpr.androidcinema.domain.get_configuration.GetConfiguration;

import retrofit2.http.GET;
import rx.Observable;

/**
 * Created by claudioribeiro on 08/07/16.
 */
public class ApiConfigurationService implements GetConfiguration.Service{

    public interface ServiceClient {
        @GET("configuration")
        Observable<ApiConfiguration> getConfiguration();
    }

    private final ServiceClient _client;

    public ApiConfigurationService(ServiceClient client){
        _client = client;
    }

    public Observable<ApiConfiguration> getConfiguration(){
        return _client.getConfiguration();
    }

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/data/api/services/ApiConfigurationService.kt
package com.crpr.androidcinema.data.api.services

import com.crpr.androidcinema.data.api.models.ApiConfiguration
import com.crpr.androidcinema.domain.common.configuration.GetConfiguration

import retrofit2.http.GET
import rx.Observable

/**
 * Created by claudioribeiro on 08/07/16.
 */
class ApiConfigurationService(private val _client: ApiConfigurationService.ServiceClient) : GetConfiguration.Service {

    interface ServiceClient {
        @get:GET("configuration")
        val configuration: Observable<ApiConfiguration>
    }

    override val configuration: Observable<ApiConfiguration>
        get() = _client.configuration

}
